function.go

```diff
@@ -46,3 +46,9 @@ func (fn *Function) NewInstance(args ...Valuer) (*Object, error) {
 	fn.ctx.deregister()
 	return getObject(fn.ctx, rtn), getError(rtn)
 }
+
+// Return the script origin for a function.
```

Comment:
This doesn't return the script origin, it returns the source map url.  It also exposes it directly on the Function type instead of through a ScriptOrigin struct.  Is there a reason why you chose to not expose a ScriptOrigin type to more closely match the original API?  Especially given that matching the original API is a project goal unless it conflicts with matching what Gophers would expect.

Reply:
Mostly introducing the whole new type to the v8go codebase and needing to figure out memory management around it led me to this implementation but open to exposing the script origin type instead and giving it a source map url getter.

Reply:
That seems like a good reason to keep it the way it is right now.

This documentation comment could still use updating though.